### ISSUE TITLE
Instrumentation of confirmation screen table is problematic

### ISSUE THREAD

**Issue by pushyamig** [tl-its-umich-edu/canvas-course-manager-next#259]
Location: Gradebook formatter, Add UM users, Create sections confirmation screen.
Impact: High/Blocker 
Type: Semantics and Structure

Issue
Instrumentation of confirmation screen table is problematic

Impact
Screen reader users will not get good information of what items have been checked since the value (checked/not checked) does not change on activation.

Keyboard-only users will not be able to make a selection because the items are not focusable.

Since each data row has a role="checkbox" - it is overriding the native semantics of a table row. Impact here is uncertain, but not complying to the DOCTYPE is potentially problematic: user agents (and assistive technology) are not required to figure out what the element "really" is.

Solution
This feels like a framework component, so we are not sure what to recommend, what is feasable. At the very least, ensure that each item is focusable (tabindex="0" rather than tabindex="-1") and manage the state of each row via aria-checked="true/false". Also ensure that there are listeners for both onclick and onkeypress. See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role

If this is not a readymade, would seriously consider either rebuilding it so that the semantics of the table are not muddled. Either give the role="checkbox" to an inline element containing the row title or rebuild as a list with inner grids for each item, and give the role="checkbox" to an inline element containing the list item title.

**Comment by ssciolla** [comment]
I'm not sure I understand the problem yet here. The items in the confirmation screen tables were not designed to be activated or checked, just reviewed. I can look again at the components referenced here, but I'm confused.

**Comment by ssciolla** [comment]
I think I've been able to resolve this (see PR #294) by removing the `checkbox` role (and an unnecessary `tabIndex={-1}`). In my testing, the table semantics now seem to be used by the screenreader.

@gsilver, do you think this would this solve the problem? Note that the confirmation table data rows are not supposed to be interactive elements, only the pagination buttons.

**Comment by gsilver** [comment]
Thanks. Making the button carry the button allows the rows to preserve the semantic value of rows.

**Comment by pushyamig** [comment]
@gsilver You opened this issue, But marked the issue to be done? Should I consider this issue to pass the test?

**Comment by melindakraft** [comment]
Moving this to Done based on @gsilver feedback in the issue.
